lib/beaker-puppet/helpers/setup/common.rb

```diff
@@ -0,0 +1,187 @@
+module Beaker
+  module DSL
+    module Helpers
+      module Setup
+        module Common
+
+          # Installs packages on the hosts.
+          #
+          # @param hosts [Array<Host>] Array of hosts to install packages to.
+          # @param package_hash [Hash{Symbol=>Array<String,Array<String,String>>}]
+          #   Keys should be a symbol for a platform in PLATFORM_PATTERNS.  Values
+          #   should be an array of package names to install, or of two element
+          #   arrays where a[0] is the command we expect to find on the platform
+          #   and a[1] is the package name (when they are different).
+          # @param options [Hash{Symbol=>Boolean}]
+          # @option options [Boolean] :check_if_exists First check to see if
+          #   command is present before installing package.  (Default false)
+          # @return true
+          def install_packages_on(hosts, package_hash, options = {})
+            platform_patterns = {
+              :redhat        => /fedora|el-|centos/,
+              :debian        => /debian|ubuntu|cumulus/,
+              :debian_ruby18 => /debian|ubuntu-lucid|ubuntu-precise/,
+              :solaris_10    => /solaris-10/,
+              :solaris_11    => /solaris-11/,
+              :windows       => /windows/,
+              :eos           => /^eos-/,
+            }.freeze
+
+            check_if_exists = options[:check_if_exists]
+            Array(hosts).each do |host|
+              package_hash.each do |platform_key,package_list|
+                if pattern = platform_patterns[platform_key]
+                  if pattern.match(host['platform'])
+                    package_list.each do |cmd_pkg|
+                      if cmd_pkg.kind_of?(Array)
+                        command, package = cmd_pkg
+                      else
+                        command = package = cmd_pkg
+                      end
+                      if !check_if_exists || !host.check_for_package(command)
+                        host.logger.notify("Installing #{package}")
+                        additional_switches = '--allow-unauthenticated' if platform_key == :debian
+                        host.install_package(package, additional_switches)
+                      end
+                    end
+                  end
+                else
+                  raise("Unknown platform '#{platform_key}' in package_hash")
+                end
+              end
+            end
+            return true
+          end
+
+          def ruby_command(host)
```

Comment:
I don't really like these here, but I was unsure where to drop them. They are utilities that are only used by tests defined in `common.rb`, `gem.rb`, `aio.rb`, etc. They are not referenced by a component when running tests, as are the rest of the methods in this file. I think it would be good to move these to a different file, but I wasn't sure the best place for that